wagtail_localize/views/edit_translation.py

```diff
@@ -872,13 +906,24 @@ def get_translation_progress(segment, locale):
                                 "wagtailadmin_pages:edit", args=[translated_instance.id]
                             )
                             if isinstance(translated_instance, Page)
-                            else reverse(
-                                "wagtailsnippets:edit",
-                                args=[
-                                    translated_instance._meta.app_label,
-                                    translated_instance._meta.model_name,
-                                    quote(translated_instance.id),
-                                ],
+                            else (
+                                reverse(
+                                    "wagtailsnippets:edit",
+                                    args=[
+                                        translated_instance._meta.app_label,
+                                        translated_instance._meta.model_name,
+                                        quote(translated_instance.id),
+                                    ],
+                                )
+                                if translated_instance._meta.model
+                                in get_snippet_models()
+                                else reverse(
+                                    "{app_label}_{model_name}_modeladmin_edit".format(
+                                        app_label=translated_instance._meta.app_label,
+                                        model_name=translated_instance._meta.model_name,
+                                    ),
+                                    args=[translated_instance.id],
```

Comment:
I feel this block is convoluted enough to warrant extracting to own method

Reply:
Extracted into a new `get_edit_url` func, also added the `get_submit_translation_url` function as well.

Reply:
ah, that reads so so much better! thank you